From 656abb5cf5a955cab56c6f10ce3f917f31c707cc Mon Sep 17 00:00:00 2001
From: Alan Tull <alan.tull@freescale.com>
Date: Wed, 6 Apr 2011 14:51:19 -0500
Subject: [PATCH 888/998] ENGR00139461-1 mxc alsa soc spdif driver

Adapt MXC ALSA S/PDIF driver to use the ASoC layer.

Signed-off-by: Alan Tull <alan.tull@freescale.com>
---
 arch/arm/configs/imx5_defconfig |    6 ++++--
 arch/arm/mach-mx5/clock.c       |    2 +-
 arch/arm/mach-mx5/devices.c     |    2 +-
 arch/arm/mach-mx5/mx53_loco.c   |    9 +++++++++
 4 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/arch/arm/configs/imx5_defconfig b/arch/arm/configs/imx5_defconfig
index dbe7025..e39aa73 100644
--- a/arch/arm/configs/imx5_defconfig
+++ b/arch/arm/configs/imx5_defconfig
@@ -1645,8 +1645,7 @@ CONFIG_SND_DRIVERS=y
 # CONFIG_SND_MTPAV is not set
 # CONFIG_SND_SERIAL_U16550 is not set
 # CONFIG_SND_MPU401 is not set
-CONFIG_SND_ARM=y
-CONFIG_SND_MXC_SPDIF=m
+# CONFIG_SND_ARM is not set
 CONFIG_SND_SPI=y
 CONFIG_SND_USB=y
 # CONFIG_SND_USB_AUDIO is not set
@@ -1656,6 +1655,7 @@ CONFIG_SND_SOC=y
 CONFIG_SND_MXC_SOC=y
 CONFIG_SND_MXC_SOC_SSI=y
 CONFIG_SND_MXC_SOC_ESAI=y
+CONFIG_SND_MXC_SOC_SPDIF_DAI=y
 CONFIG_SND_MXC_SOC_IRAM=y
 CONFIG_SND_SOC_IMX_3STACK_SGTL5000=y
 # CONFIG_SND_SOC_IMX_3STACK_AK4647 is not set
@@ -1663,11 +1663,13 @@ CONFIG_SND_SOC_IMX_3STACK_SGTL5000=y
 # CONFIG_SND_SOC_IMX_3STACK_AK5702 is not set
 # CONFIG_SND_SOC_IMX_3STACK_BLUETOOTH is not set
 CONFIG_SND_SOC_IMX_3STACK_CS42888=y
+CONFIG_SND_SOC_IMX_SPDIF=y
 # CONFIG_MXC_SSI_DUAL_FIFO is not set
 CONFIG_SND_SOC_I2C_AND_SPI=y
 # CONFIG_SND_SOC_ALL_CODECS is not set
 CONFIG_SND_SOC_CS42888=y
 CONFIG_SND_SOC_SGTL5000=y
+CONFIG_SND_SOC_MXC_SPDIF=y
 # CONFIG_SOUND_PRIME is not set
 CONFIG_HID_SUPPORT=y
 CONFIG_HID=y
diff --git a/arch/arm/mach-mx5/clock.c b/arch/arm/mach-mx5/clock.c
index b940e01..3069d05 100644
--- a/arch/arm/mach-mx5/clock.c
+++ b/arch/arm/mach-mx5/clock.c
@@ -4397,7 +4397,7 @@ static struct clk_lookup lookups[] = {
 	_REGISTER_CLOCK(NULL, "emi_intr_clk.0", emi_intr_clk[0]),
 	_REGISTER_CLOCK(NULL, "emi_intr_clk.1", emi_intr_clk[1]),
 	_REGISTER_CLOCK(NULL, "spdif_xtal_clk", spdif_xtal_clk),
-	_REGISTER_CLOCK("mxc_alsa_spdif.0", NULL, spdif0_clk[0]),
+	_REGISTER_CLOCK("mxc_spdif.0", NULL, spdif0_clk[0]),
 	_REGISTER_CLOCK("mxc_vpu.0", NULL, vpu_clk[0]),
 	_REGISTER_CLOCK(NULL, "lpsr_clk", lpsr_clk),
 	_REGISTER_CLOCK("mxc_rtc.0", NULL, rtc_clk),
diff --git a/arch/arm/mach-mx5/devices.c b/arch/arm/mach-mx5/devices.c
index e3cc1aa..ae02d84 100644
--- a/arch/arm/mach-mx5/devices.c
+++ b/arch/arm/mach-mx5/devices.c
@@ -960,7 +960,7 @@ static struct resource spdif_resources[] = {
 };
 
 struct platform_device mxc_alsa_spdif_device = {
-	.name = "mxc_alsa_spdif",
+	.name = "mxc_spdif",
 	.id = 0,
 	.num_resources = ARRAY_SIZE(spdif_resources),
 	.resource = spdif_resources,
diff --git a/arch/arm/mach-mx5/mx53_loco.c b/arch/arm/mach-mx5/mx53_loco.c
index 09bf1a2..a668a2d 100644
--- a/arch/arm/mach-mx5/mx53_loco.c
+++ b/arch/arm/mach-mx5/mx53_loco.c
@@ -568,6 +568,14 @@ static struct mxc_spdif_platform_data mxc_spdif_data = {
 	.spdif_clk = NULL,	/* spdif bus clk */
 };
 
+static struct mxc_audio_platform_data spdif_audio_data = {
+	.ext_ram = 1,
+};
+
+static struct platform_device mxc_spdif_audio_device = {
+	.name = "imx-spdif-audio-device",
+};
+
 static void mx53_loco_usbh1_vbus(bool on)
 {
 	if (on)
@@ -794,6 +802,7 @@ static void __init mxc_board_init(void)
 				ARRAY_SIZE(mxc_i2c1_board_info));
 
 	mxc_register_device(&mxc_sgtl5000_device, &sgtl5000_data);
+	mxc_register_device(&mxc_spdif_audio_device, &spdif_audio_data);
 	mx5_usb_dr_init();
 	mx5_set_host1_vbus_func(mx53_loco_usbh1_vbus);
 	mx5_usbh1_init();
-- 
1.7.4.1

